cloud/aws/providerconfig/types.go

```diff
@@ -14,22 +14,151 @@
 package providerconfig
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// AWSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderConfig field
+// for an AWS instance.
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 type AWSMachineProviderConfig struct {
```

Comment:
do we want to include an optional KeyName for ssh?

Reply:
definitely, thx!

Reply:
PR updated